pkg/controller/node/nodecontroller.go

```diff
@@ -371,6 +402,42 @@ func (nc *NodeController) Run() {
 			}
 		}, nc.nodeMonitorPeriod, wait.NeverStop)
 
+		go wait.Until(func() {
+			nc.evictorLock.Lock()
```

Comment:
I suggest the following. Remove rate limiting from IntorelablePodEvictor and TerminationEvictor (just inject FakeAllowAlways rate limiter there). Rate limiting will still be done at PodEvictor (Node) level, so it'll still be safe. After NC will decide that it's time to evict stuff from Node, there's no point in having additional rate limiting (or is there one - @smarterclayton).

Once you do that you can slightly change the semantics of values in the queues - value.Value can hold PodName instead of NodeName. NodeUID won't be needed anymore, so value.UID (your message here) could be a just a PodNamespace string. WDYT?